File: melt.py
# -*- coding: utf-8 -*-
"""
Created on Mon Jun 10 08:37:32 2024

@author: Bornand
"""

# toy research project: glacier mass balance

# melt rate

# signature: (T, melt_factor) -> meltrate (m/d)

def melt(T, melt_factor):
    """
    Calculate the glacier melt rate based on temperature and melt factor.

    Parameters:
    T (float): Temperature at a given altitude and time.
    melt_factor (float): Melt factor to scale the melt rate.

    Returns:
    float: Melt rate.
    """
    if T >= 0:
        return melt_factor * T
    else:
        return 0


## Accumulation rate function
def accumulate(T, P, T_threshold):
    """
    Calculate the accumulation on the glacier based on temperature, a temperature threshold and precipitation.

    Parameters:
    T (float): Temperature at a given altitude and time.
    P (float): Precipitation on the glacier.
    T_threshold (float): Temperature threshold below which the snow starts accumulating the precipitation
    Returns:
    float: accumulation.
    """
  if T <= T_threshold :
    return P
  else: 
    return 0



# # Example usage:
# temperature = 5  # Example temperature in degrees Celsius
# melt_factor = 0.1  # Example melt factor
# melt_rate = melt(temperature, melt_factor)
# print(f"Melt rate: {melt_rate}")



def lapse(T, dz, lapse_rate):
      """
    Calculate the glacier lapse rate based on temperature.

    Parameters:
    T (float): Temperature at a given altitude and time.
    lapse_rate: 

    Returns:
    float: lapsed temperature
    """  
    
    lapsed_T = lapse_rate*dz + T
    return lapsed_T

File: src/melt.R
## GLACIER MASS BALANCE - Functions

## Melt rate function
melt <- function(T, melt_factor){
  if(T>=0){
    meltrate <- melt_factor*T
  } else {
    meltrate <- 0
  }
  return(meltrate)
}

## Accumulation rate function
accumulate <- function(T, P, T_threshold){
  if(T<=T_threshold){
    accumulation <- P
  } else {
    accumulation <- 0
  }
  return(accumulation)
}

## Lapsed-temperature function
lapse <- function(T, dz, lapse_rate){
  lapsed_T <- lapse_rate*dz + T
  return(lapsed_T)
}

## Synthetic temperature function
temp <- function(t){
  synth_T <- -10*cos(2*pi/364 * t) - 8*cos(2*pi*t) + 5
  return(synth_T)
}

## Synthetic precipitation function
precip <- function(t){
  return(8e-3)
}



